lib/start-button.js

```diff
@@ -0,0 +1,241 @@
+    // https://blog.mozilla.org/addons/2014/03/06/australis-for-add-on-developers-2/
+    CustomizableUI.createWidget({
+      id: startButtonId,
+      type: "custom",
+      defaultArea: AREA_NAVBAR,
+      allowedAreas: [AREA_PANEL, AREA_NAVBAR],
+      onBuild: this.onBuild.bind(this)
+    });
+  },
+
+  shutdown: function(reason) {
+    CustomizableUI.destroyWidget(startButtonId);
+  },
+
+  onBuild: function(doc) {
+    Trace.sysout("startButton.onBuild;", doc);
+
+    // Create a toolbar button with associated context menu.
+    let button = new ToolbarButton({
+      document: doc,
+      id: startButtonId,
+      label: "firequery.startButton.title",
+      tooltiptext: "firequery.startButton.tip",
+      type: "menu-button",
+      "class": "toolbarbutton-1 chromeclass-toolbar-additional",
+      image: "chrome://firequery/skin/icon16.png",
+      items: this.getMenuItems.bind(this, doc.defaultView),
+      command: this.onToggleFireQuery.bind(this)
+    });
+
+    return button.button;
+  },
+
+  getAnchor: function(doc) {
+    let startButton = this.getButton(doc);
+    return new StartButtonAnchor(startButton);
+  },
+
+  getButton: function(doc) {
+    return doc.getElementById(startButtonId);
+  },
+
+  // Menu Actions
+
+  getMenuItems: function(win) {
+    Trace.sysout("start-button:getMenuItems; " + win);
+    let items = [];
+
+
+    items.push({
+      nol10n: true,
+      label: Locale.$STR("firequery.menu.visitHomePage.label"),
+      tooltiptext: Locale.$STR("firequery.menu.visitHomePage.tip"),
+      command: this.onVisitHomePage.bind(this)
+    });
+
+    items.push({
+      nol10n: true,
+      label: Locale.$STR("firequery.menu.reportIssue.label"),
+      tooltiptext: Locale.$STR("firequery.menu.reportIssue.tip"),
+      command: this.onReportIssue.bind(this)
+    });
+
+    items.push({
+      nol10n: true,
+      label: Locale.$STR("firequery.menu.group.label"),
+      tooltiptext: Locale.$STR("firequery.menu.group.tip"),
+      command: this.onDiscussionGroup.bind(this)
+    });
+
+    items.push("-");
+
+    items.push({
+      nol10n: true,
+      label: Locale.$STR("firequery.menu.about.label") + " " + self.version,
+      tooltiptext: Locale.$STR("firequery.menu.about.tip"),
+      image: "chrome://firequery/skin/logo_16x16.png",
+      command: this.onAbout.bind(this)
+    });
+
+    return items;
+  },
+
+  onToggleFireQuery: function() {
+    Trace.sysout("startButton.onToggleFireQuery;");
+
+    return getToolboxWhenReady().then(toolbox => {
+      let context = ToolboxChrome.getContext(toolbox);
+      let overlayId = ToolboxOverlay.prototype.overlayId;
+      let overlay = context.getOverlay(overlayId);
+      if (!overlay) {
+        TraceError.sysout("startButton.onToggleFireQuery; ERROR " +
+          " No overlay: " + overlayId, context);
+        return;
+      }
+
+      let win = overlay.toggleInspector();
```

Comment:
Another left over from RDPi?

Btw. what the onToggleFireQuery() method is supposed to do?

(E.g. the RDPi opens the console, but FireQuery doesn't have any extra UI)

Honza